support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateSalesforceContact.scala

```diff
@@ -32,20 +35,123 @@ class CreateSalesforceContact extends ServicesHandler[CreateSalesforceContactSta
+    ), requestId, user, product, analyticsInfo, None, None, acquisitionData)
+
+  def toNextDSRedemption(
+    product: DigitalPack,
+    redemptionData: RedemptionData
+  ): CreateZuoraSubscriptionWrapperState =
+    CreateZuoraSubscriptionWrapperState(CreateZuoraSubscriptionDigitalSubscriptionGiftRedemptionState(
+      user.id,
+      product,
+      redemptionData,
+    ), requestId, user, product, analyticsInfo, None, None, acquisitionData)
+
+  def toNextDSCorporate(
+    salesforceContactRecord: SalesforceContactRecord,
+    product: DigitalPack,
+    redemptionData: RedemptionData
+  ): CreateZuoraSubscriptionWrapperState =
+    CreateZuoraSubscriptionWrapperState(CreateZuoraSubscriptionDigitalSubscriptionCorporateRedemptionState(
+      product,
+      redemptionData,
+      salesforceContactRecord,
+    ), requestId, user, product, analyticsInfo, None, None, acquisitionData)
+
+  def toNextWeekly(
+    salesforceContactRecords: SalesforceContactRecords,
+    product: GuardianWeekly,
+    purchase: PaymentMethod
+  ): CreateZuoraSubscriptionWrapperState =
+    CreateZuoraSubscriptionWrapperState(CreateZuoraSubscriptionGuardianWeeklyState(
+      user,
+      giftRecipient.map(_.asWeekly.get),
+      product,
+      purchase,
+      firstDeliveryDate.get,
+      promoCode,
+      salesforceContactRecords,
+    ), requestId, user, product, analyticsInfo, firstDeliveryDate, promoCode, acquisitionData)
+
+  def toNextPaper(
+    salesforceContactRecord: SalesforceContactRecord,
+    product: Paper,
+    purchase: PaymentMethod
+  ): CreateZuoraSubscriptionWrapperState =
+    CreateZuoraSubscriptionWrapperState(CreateZuoraSubscriptionPaperState(
+      user,
+      product,
+      purchase,
+      firstDeliveryDate.get,
+      promoCode,
+      salesforceContactRecord,
+    ), requestId, user, product, analyticsInfo, firstDeliveryDate, promoCode, acquisitionData)
+
+  def toNextDSGift(
+    salesforceContactRecords: SalesforceContactRecords,
+    product: DigitalPack,
+    purchase: PaymentMethod
+  ): CreateZuoraSubscriptionWrapperState =
+    CreateZuoraSubscriptionWrapperState(CreateZuoraSubscriptionDigitalSubscriptionGiftPurchaseState(
+      user.billingAddress.country,
+      giftRecipient.flatMap(_.asDigitalSubscriptionGiftRecipient).get,
+      product,
+      purchase,
+      promoCode,
+      salesforceContactRecords,
+    ), requestId, user, product, analyticsInfo, firstDeliveryDate, promoCode, acquisitionData)
+
+  def toNextDSDirect(
+    salesforceContactRecord: SalesforceContactRecord,
+    product: DigitalPack,
+    purchase: PaymentMethod
+  ): CreateZuoraSubscriptionWrapperState =
+    CreateZuoraSubscriptionWrapperState(CreateZuoraSubscriptionDigitalSubscriptionDirectPurchaseState(
+      user.billingAddress.country,
+      product,
+      purchase,
+      promoCode,
+      salesforceContactRecord,
+    ), requestId, user, product, analyticsInfo, firstDeliveryDate, promoCode, acquisitionData)
+
```

Comment:
the file below (CreateZuoraSubscription.scala) is not rendered bydefault, but that is the major change in this PR! so be sure to look carefully.

Reply:
![image](https://user-images.githubusercontent.com/7304387/113931258-2a22d780-97ea-11eb-969f-0d0c0be10c2d.png)